actioncable/test/connection/client_socket_test.rb

```diff
@@ -67,16 +67,22 @@ def open_connection
       env = Rack::MockRequest.env_for "/test",
         "HTTP_CONNECTION" => "upgrade", "HTTP_UPGRADE" => "websocket",
         "HTTP_HOST" => "localhost", "HTTP_ORIGIN" => "http://rubyonrails.com"
-      io = \
+      io, client_io = \
         begin
-          Socket.pair(Socket::AF_UNIX, Socket::SOCK_STREAM, 0).first
+          Socket.pair(Socket::AF_UNIX, Socket::SOCK_STREAM, 0)
         rescue
           StringIO.new
         end
       env["rack.hijack"] = -> { env["rack.hijack_io"] = io }
 
       Connection.new(@server, env).tap do |connection|
         connection.process
+        if client_io
+          # Make sure server returns handshake response
+          Timeout.timeout(1) do
```

Comment:
Is this timeout needed?

Reply:
Yes. The test server may return nothing because of a newly introduced bug in future.

Reply:
@kamipo Do you disagree with this timeout?
If we never make such mistake, it isn't needed but I think it should be.

Reply:
Basically I don't think that we should ignore/suppress any bug signals.
If the timeout will be ignore/suppress any bugs, can we notice that?

Reply:
> If the timeout will be ignore/suppress any bug

No, the timeout doesn't ignore a bug.
It exists to notice a bug.
`client_io.readline` will block forever so we can't find which test is failed without this timeout.

Reply:
Ah, I misunderstood. The client should be able to read the handshake response right away, the timeout will let us know that it was not so. Seems reasonable.